deploy/runtools/firesim_topology_with_passes.py

```diff
@@ -191,33 +185,22 @@ def pass_no_net_host_mapping(self) -> None:
         servers = self.firesimtopol.get_dfs_order_servers()
         serverind = 0
 
-        for f1_16x in self.run_farm.f1_16s:
-            for x in range(f1_16x.get_num_fpga_slots_max()):
-                f1_16x.add_simulation(servers[serverind])
-                serverind += 1
-                if len(servers) == serverind:
-                    return
-        for f1_4x in self.run_farm.f1_4s:
-            for x in range(f1_4x.get_num_fpga_slots_max()):
-                f1_4x.add_simulation(servers[serverind])
-                serverind += 1
-                if len(servers) == serverind:
-                    return
-        for f1_2x in self.run_farm.f1_2s:
-            for x in range(f1_2x.get_num_fpga_slots_max()):
-                f1_2x.add_simulation(servers[serverind])
+        while len(servers) > serverind:
+            # this call will error if no such instances are available.
+            instance_type = self.run_farm.mapper_get_min_sim_host_inst_type_name(1)
```

Comment:
Again, better names always recommended. Basically the function is "Return a run host name (i.e. `f1.2xlarge` (in AWS's case) or `192.168.1.1` (in the IP address run farm case)) that is able to host `num_sims` simulation slots. If there are multiple hosts that can host `num_sims` slots, return the smallest host (i.e. the host with the smallest number of sim. slots".

In any case, I think the function should return what we call a "handle" (since this "handle" represents really depends on the underlying run farm). In the case of the AWS run farm a "handle" is an instance name (`f1.2xlarge`). In the case of a self managed run farm it is the IP address of a host. In other cases, it could potentially be something else (an object?). So maybe the function is better named `get_smallest_sim_host_handle(min_sim_supported=N)` (again probably needs more work to name nicely)?